src/NUnitEngine/nunit.engine.api/RuntimeType.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace NUnit.Engine
+{
+    /// <summary>
+    /// Enumeration identifying a common language
+    /// runtime implementation.
+    /// </summary>
+    public enum RuntimeType
+    {
+        /// <summary>Any supported runtime framework</summary>
+        Any,
+        /// <summary>Microsoft .NET Framework</summary>
+        Net,
+        /// <summary>Microsoft .NET Compact Framework</summary>
+        NetCF,
+        /// <summary>Microsoft Shared Source CLI</summary>
+        SSCLI,
+        /// <summary>Mono</summary>
+        Mono,
+        /// <summary>Silverlight</summary>
+        Silverlight,
+        /// <summary>MonoTouch</summary>
+        MonoTouch
```

Comment:
OK... The thing is that the RuntimeType enum is in both the engine and the framework, although it is not in our common code. All the values are actually used in the framework when processing PlatformAttribute. Of course, this means they are __not__ used in the portable build of the framework.

In the engine, we use Any, Net, Mono in a major way - those being the platforms on which the engine runs. NetCF and Silverlight are used by RuntimeFramework internally, but the engine never creates a RuntimeFramework with those values. SSCLI and MonoTouch are not used at all.

Since this has to be in both the engine and the framework  - and possibly in some of your runners - I think a magic string is the best choice, like "MONO" or "NET". That's how we handle the names of package settings already. If a new platform comes along, we can create a name for it and it can be added separately into each independent component that needs it.

For now, I would simply expose the string name in the interface and keep the internal enum. After the release, we may want to clean up internally as well.

What do you think?